pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -885,38 +935,41 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
 	driverCombo := a.checkVolumeDriverCombination(backup.Status.Volumes)
 	// If the driver combination of volumes onlykdmp or mixed of both kdmp and non-kdmp, call post exec rule
 	// backup of volume is success.
-	if driverCombo == kdmpDriverOnly || driverCombo == mixedDriver {
-		// Let's kill the pre-exec rule pod here so that application specific
-		// data  stream freezing logic works. Certain app actually unleash the WRITE when session ends.
-		// At this point we are dead sure that volume snapshot for all PVCs in the APP is done.. if not then
-		// there is issue...
-		// For detail refer pb-3823
-		for _, channel := range terminationChannels {
-			logrus.Infof("Sending termination commands to kill pre-exec pod in kdmp or mixed driver path")
-			channel <- true
-		}
-		//terminationChannels = nil
-		if backup.Spec.PostExecRule != "" {
-			log.ApplicationBackupLog(backup).Infof("Starting post-exec rule for kdmp and mixed driver path")
-			err = a.runPostExecRule(backup)
-			if err != nil {
-				message := fmt.Sprintf("Error running PostExecRule for kdmp and mixed driver scenario: %v", err)
-				log.ApplicationBackupLog(backup).Errorf(message)
-				a.recorder.Event(backup,
-					v1.EventTypeWarning,
-					string(stork_api.ApplicationBackupStatusFailed),
-					message)
-
-				backup.Status.Stage = stork_api.ApplicationBackupStageFinal
-				backup.Status.FinishTimestamp = metav1.Now()
-				backup.Status.LastUpdateTimestamp = metav1.Now()
-				backup.Status.Status = stork_api.ApplicationBackupStatusFailed
-				backup.Status.Reason = message
-				err = a.client.Update(context.TODO(), backup)
+	if !a.execRulesCompleted[string(backup.UID)] {
+		if driverCombo == kdmpDriverOnly || driverCombo == mixedDriver {
+			// Let's kill the pre-exec rule pod here so that application specific
+			// data  stream freezing logic works. Certain app actually unleash the WRITE when session ends.
+			// At this point we are dead sure that volume snapshot for all PVCs in the APP is done.. if not then
+			// there is issue...
+			// For detail refer pb-3823
+			for _, channel := range terminationChannels {
+				logrus.Infof("Sending termination commands to kill pre-exec pod in kdmp or mixed driver path")
+				channel <- true
+			}
+			//terminationChannels = nil
```

Comment:
Remove the commented code.

Reply:
Sure

Reply:
This commented code is an existing one not part of the this PR change.